tasks/e2e-yarn-ws.sh

```diff
@@ -0,0 +1,138 @@
+function exists {
+  for f in $*; do
+    test -e "$f"
+  done
+}
+
+# Exit the script with a helpful error message when any error is encountered
+trap 'set +x; handle_error $LINENO $BASH_COMMAND' ERR
+
+# Cleanup before exit on any termination signal
+trap 'set +x; handle_exit' SIGQUIT SIGTERM SIGINT SIGKILL SIGHUP
+
+# Echo every command being executed
+set -x
+
+# Go to root
+cd ..
+root_path=$PWD
+
+if hash npm 2>/dev/null
+then
+  npm i -g npm@latest
+  npm cache clean || npm cache verify
+fi
+
+# Bootstrap create-react-app monorepo
+yarn
+
+# Start local registry
+tmp_registry_log=`mktemp`
+nohup npx verdaccio@2.7.2 &>$tmp_registry_log &
+# Wait for `verdaccio` to boot
+grep -q 'http address' <(tail -f $tmp_registry_log)
+
+# Set registry to local registry
+npm set registry "$custom_registry_url"
+yarn config set registry "$custom_registry_url"
+
+# Login so we can publish packages
+npx npm-cli-login@0.0.10 -u user -p password -e user@example.com -r "$custom_registry_url" --quotes
+
+git clean -df
+./tasks/publish.sh --yes --force-publish=* --skip-git --cd-version=prerelease --exact --npm-tag=latest
+
+function verifyTest {
+  CI=true yarn test --watch=no --json --outputFile testoutput.json || return 1
+  grep -F "\"numFailedTestSuites\":0" testoutput.json -q || return 1
+  grep -F "\"numFailedTests\":0" testoutput.json -q || return 1
+  grep -F "\"numPassedTestSuites\":3" testoutput.json -q || return 1
+  grep -F "\"numPassedTests\":3" testoutput.json -q || return 1
+}
+
+function verifyBuild {
+  yarn build || return 1
+  grep -F -R --exclude=*.map "YarnWS-CraApp" build/ -q || return 1
+}
+
+# ******************************************************************************
+# Set up the apps monorepo
+# ******************************************************************************
+# Start with base workspace
+cp -r "$root_path/packages/react-scripts/fixtures/yarn-ws/ws" "$temp_app_path"
+pushd "$temp_app_path/ws"
+# Bootstrap the apps monorepo
+yarn
+
+# ******************************************************************************
+# Test CRA-App1
+# ******************************************************************************
+pushd packages/cra-app1
+yarn start --smoke-test
+verifyBuild
+verifyTest
+
+# ******************************************************************************
+# Test eject
+# ******************************************************************************
+echo yes | npm run eject
+yarn start --smoke-test
+verifyBuild
+verifyTest
+popd
+
+# ******************************************************************************
+# Test create-react-app inside workspace
+# ******************************************************************************
+# npx create-react-app --internal-testing-template="$root_path"/packages/react-scripts/fixtures/yarn-ws/ws/cra-app1 cra-app2
+# -- above needs https://github.com/facebookincubator/create-react-app/pull/3435 to user create-react-app
```

Comment:
Can you remind me the difference between what this PR and what #3435 is solving?

Does their behavior clash at all in any way, or are they complementary? Would you work on updating #3435 next if we get this in?

Reply:
#3435 allows you to run create-react-app itself, ie, to create a new app, inside an existing monorepo.  (A few parts of create-react-app assume that react-scripts is in the app's node_modules; but in a monorepo, that assumption is not valid; 3435 fixes that assumption.)  But, the app you created wouldn't be able to use shared cra source.

This PR is what allows cra-compatible source to be shared in the monorepo.

I believe #3435 is ready, and it can come right after this.  (It could come before or without this PR, too, but it would need one extra fix which I mentioned in that PR: https://github.com/facebookincubator/create-react-app/pull/3435#issuecomment-356405378)

Reply:
This makes sense, thank you.